funfact/lang/interpreter/_index_propagation.py

```diff
@@ -51,16 +51,16 @@ def neg(self, x: Numeric, **kwargs):
 
     @as_payload
     def mul(self, lhs: Numeric, rhs: Numeric, **kwargs):
-        return set(lhs.live_indices).symmetric_difference(rhs.live_indices)
+        return list(set(lhs.live_indices).symmetric_difference(rhs.live_indices))
```

Comment:
Could you please dig out the old implementation, and wrap it as a method called `ordered_symmetric_difference`, so that we can preserve the 'natural' order of the indices?